Bottom: 05605b0db6a16910c8d9eb1029c319498f6b57a4
Top:    1b18f4b592172a084c9f4497795b2e578961acba
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-12-27 17:55:58 +0000

Refresh of draw-c-opencode-the-mask-layer

---

diff --git a/src/draw.c b/src/draw.c
index 5133522..15c462d 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -871,18 +871,13 @@ DrawMaskBoardArea (int mask_type, const BoxType *drawn_area)
                                            drawn_area->X2, drawn_area->Y2);
 }
 
-struct poly_info {
-  const const BoxType *drawn_area;
-  LayerType *layer;
-};
-
 static int
 mask_poly_callback (const BoxType * b, void *cl)
 {
   struct poly_info *i = cl;
   PolygonType *polygon = (PolygonType *)b;
 
-  dapi->draw_poly (polygon, i->drawn_area, NULL);
+  gui->graphics->draw_pcb_polygon (Output.pmGC, polygon, i->drawn_area);
   return 1;
 }
 
@@ -891,7 +886,7 @@ mask_line_callback (const BoxType * b, void *cl)
 {
   LineType *line = (LineType *)b;
 
-  dapi->draw_line (line, NULL, NULL);
+  gui->graphics->draw_pcb_line (Output.pmGC, line);
   return 1;
 }
 
@@ -900,7 +895,7 @@ mask_arc_callback (const BoxType * b, void *cl)
 {
   ArcType *arc = (ArcType *)b;
 
-  dapi->draw_arc (arc, NULL, NULL);
+  gui->graphics->draw_pcb_arc (Output.pmGC, arc);
   return 1;
 }
 
@@ -931,15 +926,11 @@ DrawMask (int side, const BoxType *screen)
   struct poly_info info;
 
   if (thin)
-    {
-      gui->graphics->set_color (Output.pmGC, PCB->MaskColor);
-      gui->graphics->set_color (Output.fgGC, PCB->MaskColor); /* NB: Required, as draw_layer() uses fgGC */
-    }
+    gui->graphics->set_color (Output.pmGC, PCB->MaskColor);
   else
     {
       DrawMaskBoardArea (HID_MASK_BEFORE, screen);
       gui->graphics->use_mask (HID_MASK_CLEAR);
-      gui->graphics->set_color (Output.fgGC, "erase"); /* NB: Required, as draw_layer() uses fgGC */
     }
 
   info.layer = Layer;
@@ -950,11 +941,11 @@ DrawMask (int side, const BoxType *screen)
   r_search (Layer->text_tree,    screen, NULL, mask_text_callback, Layer);
 
   r_search (PCB->Data->pin_tree, screen, NULL, clearPin_callback, NULL);
-  r_search (PCB->Data->via_tree, screen, NULL, clearVia_callback, NULL);
+  r_search (PCB->Data->via_tree, screen, NULL, clearPin_callback, NULL);
   r_search (PCB->Data->pad_tree, screen, NULL, clearPad_callback, &side);
 
   if (thin)
-    gui->graphics->set_color (Output.pmGC, "erase"); /* NB: Only need to set pmGC back, fgGC is set when required */
+    gui->graphics->set_color (Output.pmGC, "erase");
   else
     {
       DrawMaskBoardArea (HID_MASK_AFTER, screen);
